### PR TITLE
JS Adagrams

### PR DESCRIPTION
**Pull request by krsmith7**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | Nested if else statements, compound logical operators
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | I used more Googling to find out if built in functions already existed. I found sites describing javascript more helpful than sites like StackOverflow.
What was a challenge you were able to overcome on this assignment? | Not understanding the data structure of a Javascript program
What is your favorite thing about learning a new programming language? | Learning things that a new language can do that others cannot
What is your least favorite thing about learning a new programming language? | Not knowing how seemingly simple things are written in the new language
Do you have any recommendations on how we could improve this project for the next cohort? | Giving an example of a whole Javascript program. Not necessarily Adagrams, but just a model of how interconnected functions and variables are layed out as one whole.

### CODE REVIEW

**Comment by tildeee** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | x, thank you for the ideas/feedback!
Small commits with meaningful commit messages | x
**Code Requirements** |
`drawLetters` method | x
Uses appropriate data structure to store the letter distribution | x
All tests for `drawLetters` pass | x
`usesAvailableLetters` method | x
All tests for `usesAvailableLetters` pass | x
`scoreWord` method | x
Uses appropriate data structure to store the letter scores | x
All tests for `scoreWord` pass | ? this is broken on my machine and I'm still working out why, does it pass on your machine?
`highestScoreFrom` method | x
Appropriately handles edge cases for tie-breaking logic | x
All tests for `highestScoreFrom` pass | x
**Overall** |

Katricia, great job on this project! Some of your comments imply that the logic doesn't work, but it seems to be working for me.

Weirdly enough, there is a different piece of code that I'm debugging on my machine.

That being said, I think your code looks great-- I think you did a great job transferring Adagrams into JavaScript syntax. Well done!

File: src/adagrams.js [hunk 1]
```diff
@@ -1,7 +1,147 @@
...
+  "T", "T", "T", "T", "T", "T",
+  "U", "U", "U", "U",
+  "V", "V",
+  "W", "W",
+  "X", "Y", "Y", "Z"
+  ],
+
+
+  drawLetters() {
+    let drawn = new Array;
```
Comment (tildeee) [inline src/adagrams.js:31]: Another way to write this would be `let drawn = [];`

File: src/adagrams.js [hunk 2]
```diff
@@ -1,7 +1,147 @@
...
+    for (let i = 0; i<10; i++) {
+      drawn.push( this.letterPool[ Math.floor( Math.random()* this.letterPool.length) ])}
```
Comment (tildeee) [inline src/adagrams.js:33]: missing semicolon!

Comment (tildeee) [reply src/adagrams.js:33]: Also, that ending curly brace `}` should probably be on a separate line

File: src/adagrams.js [hunk 3]
```diff
@@ -1,7 +1,147 @@
...
+
+  usesAvailableLetters(input, lettersInHand) {
+    // const setWord = new Set(word);
+    // const setDrawLetters = new Set(drawLetters);
+    //
+    // const intersection = [...setWord].filter(x=> setDrawLetters.has(x));
+    // return intersection.sort == drawLetters.sort;
+
+    let wordLetters = input.split("");
+    let drawnLetters = [...lettersInHand];
```
Comment (tildeee) [inline src/adagrams.js:46]: Nice use of the spread operator!

File: src/adagrams.js [hunk 4]
```diff
@@ -1,7 +1,147 @@
...
+    // let drawnLetters = new Set(drawLetters);
+
+    for(let i = 0; i < input.length; i++) {
+      if (drawnLetters.includes(wordLetters[i])) {
+        let index = drawnLetters.indexOf(wordLetters[i]);
```
Comment (tildeee) [inline src/adagrams.js:51]: index can be a `const` in this case

File: src/adagrams.js [hunk 5]
```diff
@@ -1,7 +1,147 @@
...
+   "Z": 10
+  },
+
+  scoreWord(word) {
+    let wordScored = word.toUpperCase().split("");
+    let score = 0;
+
+    if (word.length > 1) {
+      for (let letter of wordScored) {
+        score += this.letterScores[letter];
```
Comment (tildeee) [inline src/adagrams.js:97]: On my machine, this syntax is breaking the test. Is this true for you? I have a feeling that it might be working for you and not working on my machine.

File: src/adagrams.js [hunk 6]
```diff
@@ -1,7 +1,147 @@
...
+        } else if (word.length != 10 && highestScorer.word.length != 10 && word.length < highestScorer.word.length) {
+          highestScorer.word = word;
+          highestScorer.score = this.scoreWord(word);
+        }
+        // } else if (word.length == highestScorer.word.length)
+        // above code already will keep first appearing word as highest?
+      }
+    }
+
+    return highestScorer
```
Comment (tildeee) [inline src/adagrams.js:139]: semicolon
